pkg/resources/view.go

```diff
@@ -89,291 +89,192 @@ func View() *schema.Resource {
-	if err = d.Set("created_on", v.CreatedOn.String()); err != nil {
+	if err = d.Set("created_on", view.CreatedOn); err != nil {
 		return err
 	}
 
-	// Want to only capture the Select part of the query because before that is the Create part of the view which we no longer care about
-
-	extractor := snowflake.NewViewSelectStatementExtractor(v.Text.String)
+	// TODO [SNOW-867235]: what do we do with these extractors (added as discussion topic)?
+	// Want to only capture the SELECT part of the query because before that is the CREATE part of the view.
+	extractor := snowflake.NewViewSelectStatementExtractor(view.Text)
 	substringOfQuery, err := extractor.Extract()
 	if err != nil {
 		return err
 	}
+
 	if err = d.Set("statement", substringOfQuery); err != nil {
 		return err
 	}
-	err = d.Set("database", v.DatabaseName.String)
-	return err
+
+	return nil
 }
 
 // UpdateView implements schema.UpdateFunc.
 func UpdateView(d *schema.ResourceData, meta interface{}) error {
-	viewID, err := viewIDFromString(d.Id())
-	if err != nil {
-		return err
-	}
-	dbName := viewID.DatabaseName
-	schema := viewID.SchemaName
-	view := viewID.ViewName
-	builder := snowflake.NewViewBuilder(view).WithDB(dbName).WithSchema(schema)
 	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
 
 	// The only way to update the statement field in a view is to perform create or replace with the new statement.
 	// In case of any statement change, create or replace will be performed with all the old parameters, except statement
 	// and copy grants (which is always set to true to keep the permissions from the previous state).
 	if d.HasChange("statement") {
-		isSecureOld, _ := d.GetChange("is_secure")
-		commentOld, _ := d.GetChange("comment")
-		tagsOld, _ := d.GetChange("tag")
-
-		if isSecureOld.(bool) {
-			builder.WithSecure()
+		oldIsSecure, _ := d.GetChange("is_secure")
+		oldComment, _ := d.GetChange("comment")
+		oldTags, _ := d.GetChange("tag")
+
+		createRequest := sdk.NewCreateViewRequest(id, d.Get("statement").(string)).
+			WithOrReplace(sdk.Bool(true)).
+			WithCopyGrants(sdk.Bool(true)).
+			WithComment(sdk.String(oldComment.(string))).
+			WithTag(getTagsFromList(oldTags.([]any)))
+
+		if oldIsSecure.(bool) {
+			createRequest.WithSecure(sdk.Bool(true))
 		}
 
-		query, err := builder.
-			WithReplace().
-			WithStatement(d.Get("statement").(string)).
-			WithCopyGrants().
-			WithComment(commentOld.(string)).
-			WithTags(getTags(tagsOld).toSnowflakeTagValues()).
-			Create()
+		err := client.Views.Create(ctx, createRequest)
 		if err != nil {
-			return fmt.Errorf("error when building sql query on %v, err = %w", d.Id(), err)
-		}
-
-		if err := snowflake.Exec(db, query); err != nil {
 			return fmt.Errorf("error when changing property on %v and performing create or replace to update view statements, err = %w", d.Id(), err)
 		}
 	}
 
 	if d.HasChange("name") {
```

Comment:
Same with this rename, maybe move it as the last update operation? Updates after rename, like comment, will throw an error

Reply:
Fixed like the previous one